apps/web/src/hooks/usePaginatedQuery.ts

```diff
@@ -34,13 +35,16 @@ export const usePaginatedQuery = <TResponse extends object>({
   paginationOptions = {},
 }: UsePaginatedQueryProps<TResponse>) => {
   const { currentPageNumber, pageSize, setCurrentPageNumber, setPageSize } = usePaginationState(paginationOptions);
+  const { search, setSearch } = useSearchState(paginationOptions);
```

Comment:
created the `useSearchState` hook that reads the `query` value from the URL query params and allows to updating it